Branch tipos de cambio

Ulises-Berasain opened the pull request.

Ulises-Berasain committed:
Divisas y fechas

Al ingresar una divisa o alguna fecha entrega los datos en una lista.

Ulises-Berasain committed:
Errores

Agrego Errores visible para que el usuario sepa que hacer y proceso de cypress

Ulises-Berasain committed:
Tipos de cambio + Cypress

Agregue una parte de Cypress para correccion automatica.

Ulises-Berasain committed:
Update index.html

fsodano commented on cypress/integration/tipos-cambio.js:
```diff
@@ -0,0 +1,31 @@
+/// <reference types="Cypress"/>
+///<reference types="jquery" />
+
+const pagina = "http://192.168.0.111:8080";
```
ojo que esto va a cambiar según la red LAN donde estés, creo que lo deberías apuntar a tu localhost

fsodano reviewed (commented):
El video no lo tenés que commitear, debería estar ignorado en .gitignore

fsodano commented on index.html:
```diff
@@ -0,0 +1,55 @@
+
+    <div class="contenedor-base">
+        <p class="titulo-base">Ingresar divisa base</p>
+        <input type="text" class="base" id="base" placeholder="USD, EUR, ..."/>
```
esto estaría bueno que sea un <select>

fsodano commented on index.html:
```diff
@@ -0,0 +1,55 @@
+
+    <div class="contenedor-fecha">
+        <p class="titulo-fecha">Ingresar fecha (año-mes-dia)</p>
+        <input type="text" class="fecha" id="fecha" placeholder="aaaa-mm-dd"/>
```
esto puede ser un type="date"

fsodano commented on package.json:
```diff
@@ -0,0 +1,31 @@
+  "version": "1.0.0",
+  "description": "Muestra de los tipos de cambio entre EUR y DOL",
+  "main": "main.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
esto no puede estar vacío, tenés que poner los scripts para correr tus tests y levantar el programa

fsodano commented on package.json:
```diff
@@ -0,0 +1,31 @@
+  "author": "Ulises Berasain",
+  "license": "ISC",
+  "devDependencies": {
+    "3": "^2.1.0",
+    "-": "0.0.1",
```
casi seguro que estos son errores

fsodano commented on package.json:
```diff
@@ -0,0 +1,31 @@
+    "cypress": "^4.12.1",
+    "http-server": "^0.12.3",
+    "jquery": "^1.12.4",
+    "json": "^9.0.6",
```
me parece que no necesitás esto

fsodano commented on src/main.js:
```diff
@@ -0,0 +1,49 @@
+      fetch(`https://api.exchangeratesapi.io/latest?base=${$($base).val()}`)
+      .then(respuesta => respuesta.json())
+      .then(respuestaJSON => {
+        $("h1").text(
```
esto estaría mejor que lo referencies por `id`

Ulises-Berasain committed:
Correcciones

Correccion cypress para que la direccion sea siempre el localhost, cambiar el input a type="date".
